src/_bentoml_impl/client/http.py

```diff
@@ -94,6 +96,13 @@ def _make_client(
             url = "http://127.0.0.1:3000"
         elif parsed.scheme == "tcp":
             url = f"http://{parsed.netloc}"
+        elif app is not None:
+            if client_cls is httpx.Client:
+                from a2wsgi import ASGIMiddleware
+
+                transport = httpx.WSGITransport(app=ASGIMiddleware(app))
```

Comment:
httpx.Client can only interact with WSGITransport while httpx.AsyncClient interacts with ASGITransport.
It depends on whether user uses sync or async client. @aarnphm